fe/fe-core/src/main/java/com/starrocks/load/ExportMgr.java

```diff
@@ -305,31 +303,32 @@ public List<List<String>> getExportJobInfosByIdOrState(
 
                 exportJobInfos.add(jobInfo);
 
-                if (++counter >= resultNum) {
+                if (isLimitBreak && ++counter >= resultNum) {
                     break;
                 }
             }
         } finally {
             readUnlock();
         }
 
-        // TODO: fix order by first, then limit
         // order by
-        ListComparator<List<Comparable>> comparator = null;
+        ListComparator<List<Comparable>> comparator;
         if (orderByPairs != null) {
             OrderByPair[] orderByPairArr = new OrderByPair[orderByPairs.size()];
-            comparator = new ListComparator<List<Comparable>>(orderByPairs.toArray(orderByPairArr));
+            comparator = new ListComparator<>(orderByPairs.toArray(orderByPairArr));
         } else {
             // sort by id asc
-            comparator = new ListComparator<List<Comparable>>(0);
+            comparator = new ListComparator<>(0);
         }
-        Collections.sort(exportJobInfos, comparator);
+        exportJobInfos.sort(comparator);
 
         List<List<String>> results = Lists.newArrayList();
-        for (List<Comparable> list : exportJobInfos) {
-            results.add(list.stream().map(e -> e.toString()).collect(Collectors.toList()));
+        for (int i = 0; i < exportJobInfos.size(); i++) {
+            if (i > resultNum - 1) {
+                break;
+            }
```

Comment:
```suggestion
        int upperBound = Math.min(resultNum, exportJobInfos.size());
        for (int i = 0; i < upperBound; i++) {
```

Reply:
Thank you for your review. I have made the necessary changes